app/models/course_session.rb

```diff
@@ -6,6 +6,8 @@ class CourseSession < ApplicationRecord
 
   scope :students, -> { joins(:course_session_participants).merge(CourseSessionParticipant.student) }
 
+  enum status: [:draft, :admissions, :active, :finished, :archived]
```

Comment:
You should probably modify the `accepting_applications` method to use this status.

Reply:
Modified.